examples/example_3d_simulate/example_3d_fit.py

```diff
@@ -1,130 +1,116 @@
-from gammapy.cube import SkyCube
-from gammapy.extern.pathlib import Path
-
-
-def load_cubes(config):
-    cube_dir = Path(config['logging']['working_dir'])
-    npred_cube = SkyCube.read(cube_dir / 'npred_cube.fits.gz')
-    exposure_cube = SkyCube.read(cube_dir / 'exposure_cube.fits', format='fermi-exposure')
-    # print(exposure_cube)
-    # print('exposure sum: {}'.format(np.nansum(exposure_cube.data)))
-    i_nan = np.where(np.isnan(exposure_cube.data))
-    exposure_cube.data[i_nan] = 0
-
-    # npred_cube_convolved = SkyCube.read(cube_dir / 'npred_cube_convolved.fits.gz')
-
-    return dict(counts=npred_cube, exposure=exposure_cube)
-
-
-def read_config(filename):
-    with open(filename) as fh:
-        config = yaml.load(fh)
-
-    # TODO: fix the following issue in a better way (e.g. raise an error or fix somehow)
-    # apparently this gets returned as string, but we want float!?
-    # prefactor1: 2e-12
-    config['model']['prefactor1'] = float(config['model']['prefactor1'])
-
-    return config
-
-
-def main():
-    config = read_config('config.yaml')
-    # ref_cube = make_ref_cube(config)
-    # target_position = SkyCoord(config['model']['ra1'], config['model']['dec1'], unit="deg").galactic
-
-    cubes = load_cubes(config)
-    print('which available cubes:', cubes)
-
-    # converting data SkyCube to sherpa-format cube
-    counts = cubes['counts'].to_sherpa_data3d() #dstype='Data3DInt')
-    print('counts: ', counts)
-    # Define a 2D gaussian for the spatial model
-    # spatial_model = NormGauss2DInt('spatial-model')
+from astropy import log
 
-    # Define a power law for the spectral model
-    # spectral_model = PowLaw1D('spectral-model')
+from gammapy.image.models import SkyGaussian2D
+from gammapy.spectrum.models import PowerLaw
+from gammapy.maps import WcsNDMap
+from gammapy.cube import SkyModel, SkyModelMapEvaluator
+from gammapy.stats import cash
+from gammapy.utils.fitting import fit_minuit
+from example_3d_simulate import get_sky_model
 
-    coord = cubes['counts'].sky_image_ref.coordinates(mode="edges")
-    energies = cubes['counts'].energies(mode='edges').to("TeV")
-    print('my energy bins: ', energies)
-    # import IPython; IPython.embed();
 
-    # Set up exposure table model
-    exposure = TableModel('exposure')
-    exposure.load(None, cubes['exposure'].data.ravel())
-    exposure.ampl.freeze()
-
-    use_psf = config['model']['use_psf']
-
-    model_gammapy = get_model_gammapy(config)
-
-    spectral_model_sherpa = model_gammapy.spectral_model.to_sherpa()
-    spectral_model_sherpa.ampl.thaw()
-
-    spatial_model_sherpa = model_gammapy.spatial_model.to_sherpa()
-    spatial_model_sherpa.xpos.freeze()
-    spatial_model_sherpa.ypos.freeze()
-    spatial_model_sherpa.r0.freeze()
-    spatial_model_sherpa.width.freeze()
+def load_cubes():
+    npred_cube = WcsNDMap.read('npred.fits')
+    exposure_cube = WcsNDMap.read('exposure.fits')
+    i_nan = np.where(np.isnan(exposure_cube.data))
```

Comment:
This shouldn't be necessary here, exposure shouldn't have `nan`.
Is this for pixels outside the max FOV offset cut?
Then we should just fill those with 0 instead of NaN.
Can you please either fix properly, or leave a TODO comment here to remember to make this change when filling exposure?

Reply:
I copied this from the old script, don't know what it does

Reply:
The exposure map is now filled properly, has no `nan` pixels:
```
>>> import numpy as np
>>> from gammapy.maps import Map
>>> m = Map.read("examples/example_3d_simulate/exposure.fits")
>>> np.isnan(m.data).sum()
0
```
So just remove these two old lines that do `nan` handling.